db-server/test/backend/db_tests.js

```diff
@@ -54,1938 +54,1313 @@ function createEmail(data) {
+        .then(() => {
+          accountData = createAccount()
+          return db.createAccount(accountData.uid, accountData)
         })
-        .then(function() {
-          return db.account(ACCOUNT.uid)
+    })
+
+    describe('account', () => {
+      let accountData
+      beforeEach(() => {
+        accountData = createAccount()
+        return db.accountExists(accountData.emailBuffer)
+          .then(() => assert.fail, (err) => assert.equal(err.code, 404, 'Not found'))
+          .then(() => db.createAccount(accountData.uid, accountData))
+      })
+
+      describe('db.createAccount', () => {
+        it('should create account', () => {
+          accountData = createAccount()
+          return db.accountExists(accountData.emailBuffer)
+            .then(() => assert.fail, (err) => assert.equal(err.code, 404, 'Not found'))
+            .then(() => db.createAccount(accountData.uid, accountData))
+            .then((account) => {
+              assert.deepEqual(account, {}, 'Returned an empty object on account creation')
+              return db.accountExists(Buffer(accountData.email))
+                .then((exists) => assert(exists, 'account exists for this email address'), assert.fail)
+            })
         })
-        .then(function(account) {
-          assert.deepEqual(account.uid, ACCOUNT.uid, 'uid')
-          assert.equal(account.email, ACCOUNT.email, 'email')
-          assert.deepEqual(account.emailCode, ACCOUNT.emailCode, 'emailCode')
-          assert.equal(!! account.emailVerified, ACCOUNT.emailVerified, 'emailVerified')
-          assert.deepEqual(account.kA, ACCOUNT.kA, 'kA')
-          assert.deepEqual(account.wrapWrapKb, ACCOUNT.wrapWrapKb, 'wrapWrapKb')
-          assert(! account.verifyHash, 'verifyHash field should be absent')
-          assert.deepEqual(account.authSalt, ACCOUNT.authSalt, 'authSalt')
-          assert.equal(account.verifierVersion, ACCOUNT.verifierVersion, 'verifierVersion')
-          assert.equal(account.createdAt, ACCOUNT.createdAt, 'createdAt')
-          assert.equal(account.verifierSetAt, account.createdAt, 'verifierSetAt has been set to the same as createdAt')
-          assert.equal(account.locale, ACCOUNT.locale, 'locale')
+
+        it('fails with duplicate account', () => {
+          return db.createAccount(accountData.uid, accountData)
+            .then(assert.fail, (err) => {
+              assert(err, 'trying to create the same account produces an error')
+              assert.equal(err.code, 409, 'error code')
+              assert.equal(err.errno, 101, 'error errno')
+              assert.equal(err.message, 'Record already exists', 'message')
+              assert.equal(err.error, 'Conflict', 'error')
+            })
         })
-        .then(function() {
-          return db.checkPassword(ACCOUNT.uid, {verifyHash: Buffer(crypto.randomBytes(32))})
+      })
+
+      describe('db.account', () => {
+        it('should return account', () => {
```

Comment:
When there is a `describe` with no `before`/`after`/`beforeEach`/`afterEach`, I tend to think the enclosed tests should just be moved up a level into the parent `describe`. The extra layer of indirection isn't adding any value.